packages/request-client.js/src/http-metamask-data-access.ts

```diff
@@ -0,0 +1,306 @@
+import { Block } from '@requestnetwork/data-access';
+import { requestHashSubmitterArtifact } from '@requestnetwork/smart-contracts';
+import { DataAccessTypes } from '@requestnetwork/types';
+import Utils from '@requestnetwork/utils';
+import axios, { AxiosRequestConfig } from 'axios';
+import { ethers } from 'ethers';
+
+// Maximum number of retries to attempt when http requests to the Node fail
+const HTTP_REQUEST_MAX_RETRY = 3;
+
+// Delay between retry in ms
+const HTTP_REQUEST_RETRY_DELAY = 100;
+
+/**
+ * Exposes a Data-Access module over HTTP
+ */
+export default class HttpMetamaskDataAccess implements DataAccessTypes.IDataAccess {
+  /**
+   * Cache block persisted directly (in case the node did not have the time to retrieve it)
+   * (public for easier testing)
+   */
+  public cache: {
+    [channelId: string]: {
+      [ipfsHash: string]: { block: DataAccessTypes.IBlock; storageMeta: any } | null;
+    };
+  } = {};
+
+  /**
+   * Configuration that will be sent to axios for each request.
+   * We can also create a AxiosInstance with axios.create() but it dramatically complicates testing.
+   */
+  private axiosConfig: AxiosRequestConfig;
+
+  private submitterContract: ethers.Contract | undefined;
+  private provider: ethers.providers.JsonRpcProvider | ethers.providers.Web3Provider;
+  private networkName: string = '';
+
+  /**
+   * Creates an instance of HttpDataAccess.
+   * @param nodeConnectionConfig Configuration options to connect to the node. Follows Axios configuration format.
+   */
+  constructor(
+    {
+      nodeConnectionConfig,
+      web3,
+      urlProvider,
```

Comment:
`web3providerUrl` ?

Reply:
It is not necessary a web3 provider, What about `ethereumProviderUrl`